### PR TITLE
Java singleton design pattern

### LINKED ISSUE #3407: Java Singleton Design Pattern
### Proposed title of article

Java Singleton Design Pattern

### Introduction paragraph (2-3 paragraphs):

Singleton is a creational design pattern that assures that only one object of its kind exists and gives other code a single point of access to it. Singletons have nearly the same advantages and disadvantages as global variables. They are quite useful, yet they violate the modularity of our code.

In this article, the reader will learn about the singleton design pattern. Understand early Instantiation of Singleton Pattern and lazy Instantiation of Singleton Pattern.

### Key takeaways: 

 In this article we will cover the following;

1.  Singleton Design Pattern
2. Eager loading singleton object
3. Lazy loading singleton object
4. Synchronized with Thread in Lazy Loading
5. Using Thread with Lazy Loading (Double Check Locking Technique)
6. Enum with Singleton Pattern

### References: 
 N/A

### PR DESCRIPTION
**Pull request by carowanjiku**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3407 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by carowanjiku** [comment]
Hi, @hectorkambow  @Dawe-7 

This is a new PR  that was reviewed by @Dawe-7 (duplicate of #3692 ).
 I have made all the changes that you suggested.
Thank you.

**Review by Dawe-7 — APPROVED** [review]
@hectorkambow this article is ready for final review.

cc: @Kayere

File: content/articles/java-singleton-design-pattern/index.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    {
+        blueprint p01 = new blueprint();
+        blueprint p02 = new blueprint();
+    }
+}
+```
+
+When you create multiple instances, the values for both reference variables will be different. As a result, it is not a Singleton class.
+
+In the example above, r1 and r2 are two reference elements with different values. If an object is null, lazy, and eager, design trends produce a new singleton entity or any specific entity.
```
Comment (Kayere) [inline content/articles/java-singleton-design-pattern/index.md:68]: I cant seem to  find r1 and r2 in you code snippet.

File: content/articles/java-singleton-design-pattern/index.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+            }
+        });
+
+        r1.start();
+        r2.start();
+    }
+}
+```
+
+In the code above, two threads, r1 and r2, are specified, both of which call the same procedure, and both are initialized at an equal moment. We should note that with thread, we may use synchronized with the `getInstance` method.
```
Comment (Kayere) [inline content/articles/java-singleton-design-pattern/index.md:246]: Clarify this sentence. Especially the last part.

File: content/articles/java-singleton-design-pattern/index.md [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+- [Synchronization of thread in lazy loading](#synchronization-of-thread-in-lazy-loading)
+- [Lazy loading using Thread double-locking technique](#lazy-loading-using-thread-double-locking-technique)
+- [Enum with Singleton pattern](#enum-with-singleton-pattern)
+
+### Overview of Singleton design pattern
+The term "singleton" refers to a component that only has one instance. By preventing the construction of a class, it guarantees that only one entity of the class resides within the Java virtual machine.
+
+In Java, a Singleton class facilitates the creation of only one instance. This one object or instance is accessible to all other classes.
+
+Fields that are unique to each case if we have such a feature like a class, such as the unique or static components, will only be used once.
```
Comment (Kayere) [inline content/articles/java-singleton-design-pattern/index.md:34]: Please clarify this sentence.

**Review by Kayere — COMMENTED** [review]
This is a  great article @carowanjiku - I have highlighted a couple sentences that I think need a little clarification. Please do so and the article will be good to go.

**Comment by carowanjiku** [comment]
Hi, @Kayere 
I have done  the requested changes.
Thank You.

**Review by Kayere — APPROVED** [review]
Congratulations for publishing your first article @carowanjiku - Please go over the changes made by your peer reviewer and I to help improve the quality of your future articles.

cc: @Dawe-7

**Pull request merged**